src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,110 @@
+package duke.storage;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Pattern;
+
+import duke.exception.DukeException;
+import duke.exception.DukeFileNotFoundException;
+import duke.exception.DukeCorruptedFileException;
+import duke.exception.DukeReadFileException;
+import duke.exception.DukeUnrecognisedTaskTypeException;
+import duke.exception.DukeWriteFileException;
+import duke.task.DeadlineTask;
+import duke.task.EventTask;
+import duke.task.Task;
+import duke.task.TodoTask;
+import duke.utils.TaskList;
+
+public class Storage {
+    public static final String DEFAULT_FILE_PATH = "data/data.txt";
+
+    private String filePath;
+
+    public Storage() {
+        this.filePath = Storage.DEFAULT_FILE_PATH;
+    }
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    public String getFilePath() {
+        return filePath;
+    }
+
+    public TaskList loadTaskList() throws DukeException {
+        TaskList tasks = new TaskList();
+        try {
+            FileReader fr = new FileReader(filePath);
+            BufferedReader br = new BufferedReader(fr);
+            String line = br.readLine();
+            while (line != null) {
+                Task newTask = parseTask(line);
+                tasks.addTask(newTask);
+                line = br.readLine();
+            }
+            br.close();
+            return tasks;
+        } catch (FileNotFoundException e) {
+            throw new DukeFileNotFoundException(e);
+        } catch (IOException e) {
+            throw new DukeReadFileException(e);
+        } catch (ArrayIndexOutOfBoundsException
+                | DukeUnrecognisedTaskTypeException
+                | DateTimeParseException e) {
+            throw new DukeCorruptedFileException(filePath);
```

Comment:
Nice job in differentiating the type of exceptions to cater for the different possible errors.